src/AdoNet/Orleans.Persistence.AdoNet/Storage/Provider/Orleans3CompatibleHasher.cs

```diff
@@ -0,0 +1,24 @@
+using Orleans.Storage;
+
+namespace Orleans.Storage
+{
+    /// <summary>
+    /// Orleans v3-compatible hasher implementation for non-string-only grain key ids.
+    /// </summary>
+    internal class Orleans3CompatibleHasher : IHasher
+    {
+        /// <summary>
+        /// <see cref="IHasher.Description"/>
+        /// </summary>
+        public string Description { get; } = $"Orleans v3 hash function ({nameof(JenkinsHash)}).";
+
+        /// <summary>
+        /// <see cref="IHasher.Hash(byte[])"/>.
+        /// </summary>
+        public int Hash(byte[] data)
```

Comment:
```suggestion
        public int Hash(ReadOnlySpan<byte> data)
```
?

Reply:
It will be a breaking change - `IHashPicker` interface is a part of the public API.

We can extend `IHashPicker` with new method `Hash(ReadOnlySpan<byte>)` with default implementation or add new interface with this method.

Reply:
It's possible to add method overload just in implementations. May be it's the best option as `byte[] data` is originally produced by `AdoGrainKey.GetHashBytes()`.